integration_test/cases/preload.exs

```diff
@@ -708,6 +709,70 @@ defmodule Ecto.Integration.PreloadTest do
     assert %User{id: ^uid1} = item.user
   end
 
+  test "preload belongs_to in embedded_schema from container schema" do
```

Comment:
Since the test is long and also not covering all cases you might want to consider the following:

1. Create a describe block for preloading embeds. Use the setup block to create all the data for the tests.
2. Create separate tests for the following cases:
    i.    parent -> embeds_one -> belongs_to
    ii.   parent -> embeds_many -> has_many
    iii.  parent -> belongs_to -> embeds_one -> belongs_to
    iv.   parent -> has_many -> embeds_many -> has_many
    v.    raising when embed is preloaded without an assoc

These will cover whether assoc can preload embed properly and vice versa.

Reply:
Regard #2:

a. I don't think it matters whether the preloaded association is `belongs_to` or `has_many` because this PR does not touch on that part of the code. This is only dealing with `embeds_one` and `embeds_many`, the actual association is loaded through the code paths that are already tested.
b. Can we keep the scope of this PR to the case where the parent contains the embed schema and then the association is within the embed? I agree that `parent -> assoc -> embed -> assoc` would be neat, but maybe that can be a separate issue?

Reply:
Took my shot at using the describe block!

Reply:
Reagrd `#2` again:

I am also a bit new so I don't know what a preload with a `parent -> assoc -> embed -> assoc` would look like, maybe you can help me with the function call for that?

Reply:
Sorry you made good points. Some of the tests I suggested are pointless. 

1. We don't care about `has_many` because we know it works already when `embed` is the parent. So only using `belongs_to` is ok.
2. `parent -> assoc -> embed -> assoc` isn't necessary because we're calling `preload_each` recursively. So it's no different than testing `parent -> embed`

So I guess just making sure `embeds_one` and `embeds_many` works as expected is enough.